src/components/join-flow/join-flow-steps.jsx

```diff
@@ -22,14 +93,81 @@ class UsernameStep extends React.Component {
-                    description={this.props.intl.formatMessage({id: 'registration.usernameStepDescription'})}
-                    title={this.props.intl.formatMessage({id: 'general.joinScratch'})}
-                    waiting={this.props.waiting}
-                />
+                {props => {
+                    const {
+                        errors,
+                        handleSubmit,
+                        isSubmitting,
+                        validateField,
+                        values
+                    } = props;
+                    return (
+                        <JoinFlowStep
+                            description={this.props.intl.formatMessage({id: 'registration.usernameStepDescription'})}
+                            title={this.props.intl.formatMessage({id: 'general.joinScratch'})}
+                            waiting={isSubmitting}
+                            onSubmit={handleSubmit}
+                        >
+                            <div>
+                                <div className="username-label">
+                                    <b>
+                                        {this.props.intl.formatMessage({id: 'registration.createUsername'})}
+                                    </b>
+                                </div>
+                                <FormikInput
+                                    className={errors.username ? 'fail' : ''}
+                                    error={errors.username}
+                                    id="username"
+                                    name="username"
+                                    validate={this.validateUsernameIfPresent}
+                                    onBlur={() => validateField('username')} // eslint-disable-line react/jsx-no-bind
+                                />
+                                <b>
+                                    {this.props.intl.formatMessage({id: 'general.password'})}
+                                </b>
+                                <div
+                                    onClick={this.handleChangeShowPassword}
+                                >
+                                    {this.state.showPassword ? 'Hide password' : (
+                                        this.props.intl.formatMessage({id: 'registration.showPassword'})
+                                    )}
+                                </div>
+                                <FormikInput
+                                    className={errors.password ? 'fail' : ''}
+                                    error={errors.password}
+                                    id="password"
+                                    name="password"
+                                    type={this.state.showPassword ? 'text' : 'password'}
+                                    validate={this.validatePasswordIfPresent}
+                                    onBlur={() => validateField('password')} // eslint-disable-line react/jsx-no-bind
```

Comment:
Why do you need to use an arrow function here (and in the other places in this file)? I know very little about this, but seems like the check is there for a reason.

Reply:
So, it's a linting rule because when you use these, they create a new function on every render. If you render often, that can be slow.

But there are times there's really no way around them, and I think this is one, because of the need to pass the 'password' param.

Also, if the component won't be rerendering much, it's not a big deal. There's even a lot of disagreement out there about if it's ever a big deal.

Reply:
Cool. Thanks for the explanation!